stickermanager.py

```diff
@@ -0,0 +1,276 @@
+Scores = NamedTuple('Scores', sum=int, yes=int, no=int)
+
+current_vote: Optional[dict] = None
+current_vote_lock: asyncio.Lock = asyncio.Lock()
+sticker_pack = None
+
+with open(os.path.join(os.path.dirname(__file__), 'stickermanager.tsv')) as f:
+    WEIGHTS = {
+        int(uid) if uid.strip().isnumeric() else uid:
+            (0 if weight == '-' else float(weight), None if name == '-' else name)
+        for uid, weight, name
+        in (line.strip().split('\t', 2)
+            for line in f
+            if len(line) > 0 and not line[0] == '#' and not line.isspace())
+    }
+
+    VOTES_REQUIRED = WEIGHTS.pop('votes required')[0]
+    DEFAULT_WEIGHT = WEIGHTS.pop('default weight')[0]
+    STICKER_PACK_TITLE = WEIGHTS.pop('sticker title')[1]
+    STICKER_PACK_SHORT_NAME = WEIGHTS.pop('sticker short')[1]
+    ADMIN_USER_ID = int(WEIGHTS.pop('sticker owner')[1])
+    ALLOWED_CHATS = [int(gid.strip())
+                     for gid in WEIGHTS.pop('allowed chats')[1].split(',')
+                     if not gid.isspace()]
+
+
+async def create_sticker_pack(bot: TelegramClient, item: InputStickerSetItem
+                              ) -> Tuple[bool, Optional[StickerSet]]:
+    global sticker_pack
+    try:
+        with open(os.path.join(os.path.dirname(__file__), 'stickermanager.json')) as file:
+            sp_data = json.load(file)
+            sticker_pack = InputStickerSetID(id=sp_data['id'], access_hash=sp_data['access_hash'])
+        return False, None
+    except (FileNotFoundError, KeyError) as e:
+        stickerset: StickerSet = await bot(CreateStickerSetRequest(
+            user_id=ADMIN_USER_ID,
+            title=STICKER_PACK_TITLE,
+            short_name=STICKER_PACK_SHORT_NAME,
+            stickers=[item]
+        ))
+        sticker_pack = InputStickerSetID(id=stickerset.set.id,
+                                         access_hash=stickerset.set.access_hash)
+        with open(os.path.join(os.path.dirname(__file__), 'stickermanager.json'), 'w') as file:
+            json.dump({
+                'id': stickerset.set.id,
+                'access_hash': stickerset.set.access_hash
+            }, file)
+        return True, stickerset
+
+
+async def add_sticker_to_pack(bot: TelegramClient) -> Tuple[StickerSet, InputDocument]:
+    global sticker_pack
+    animated = current_vote['animated']
+    if animated:
+        file = await bot.upload_file(current_vote['filepath'])
+        mime = 'application/x-tgsticker'
+    else:
+        img = Image.open(current_vote['filepath'])
+        w, h = img.size
+        if w > 512 or h > 512:
+            img.thumbnail((512, 512), Image.ANTIALIAS)
+        else:
+            if w > h:
+                img = img.resize((512, int(h * (512 / w))), Image.ANTIALIAS)
+            elif w < h:
+                img = img.resize((int((w * (512 / h))), 512), Image.ANTIALIAS)
+            else:
+                img = img.resize((512, 512), Image.ANTIALIAS)
```

Comment:
This is the same case as the first `if`. Probably the `w > h` and `w < h` branches can go first, *and then* have this be the last one.